apps/manager_app/auth_card.c

```diff
@@ -273,66 +225,15 @@ static manager_error_code_t handle_auth_card_query(
 /*****************************************************************************
  * STATIC FUNCTIONS
  *****************************************************************************/
-static manager_error_code_t get_and_decode_valid_query_from_host(
-    manager_query_t *query_out) {
-  evt_status_t evt_status = get_events(EVENT_CONFIG_USB, TIMEOUT_SELECTION);
-
-  HANDLE_P0_EVENTS(evt_status.p0_event);
-
-  if (false == evt_status.usb_event.flag) {
-    /* We don't expect this to happen, either P0 or USB event must occur to
-     * return from get events*/
-    return MANAGER_TASK_FAILED;
-  }
-
-  if (!decode_manager_query(evt_status.usb_event.p_msg,
-                            evt_status.usb_event.msg_size,
-                            query_out)) {
-    return MANAGER_TASK_DECODING_FAILED;
-  }
-
-  if ((query_out->which_request != MANAGER_QUERY_AUTH_CARD_TAG)) {
-    return MANAGER_TASK_UNKNOWN_QUERY_REQUEST;
-  }
-
-  return MANAGER_TASK_SUCCESS;
-}
-
-static manager_error_code_t send_auth_card_response(
-    manager_auth_card_response_t *resp) {
-  // Initialize respose with zero values
-  manager_result_t result = MANAGER_RESULT_INIT_ZERO;
-
-  /* TODO: Update the buffer size when all option files have been updated for
-   * manager app*/
-  uint8_t buffer[MANAGER_AUTH_CARD_RESPONSE_SIZE + 4] = {0};
-  size_t bytes_written = 0;
-
-  // Set auth card valid tags
-  result.which_response = MANAGER_QUERY_AUTH_CARD_TAG;
-
-  // Copy response to pb result struct
-  memcpy(&(result.auth_card), resp, sizeof(manager_auth_card_response_t));
-
-  // Encode result to byte-stream
-  if (false ==
-      encode_manager_result(&result, buffer, sizeof(buffer), &bytes_written)) {
-    // Encoding failed, return false
-    return MANAGER_TASK_ENCODING_FAILED;
-  }
-
-  usb_send_msg(buffer, bytes_written);
-  return MANAGER_TASK_SUCCESS;
-}
-
 static manager_error_code_t prepare_card_auth_context(
     auth_card_data_t *auth_card_data) {
   auth_card_data->ctx.acceptable_cards = ACCEPTABLE_CARDS_ALL;
 
   if (auth_card_data->query->auth_card.initiate.has_card_index) {
     if (4 < auth_card_data->query->auth_card.initiate.card_index ||
         0 == auth_card_data->query->auth_card.initiate.card_index) {
-      /* TODO: Return invalid query error code to host*/
+      manager_send_error(ERROR_COMMON_ERROR_CORRUPT_DATA_TAG,
+                         ERROR_DATA_FLOW_INVALID_DATA);
       return MANAGER_TASK_DECODING_FAILED;
```

Comment:
Can you verify if this return code is appropriate?

Reply:
Earlier the idea for these return codes was that all errors would be reported based on these error codes, but now as we are sending the errors directly, we can remove all the error codes and keep bool instead.

Reply:
removed